### PR TITLE
Allow flags in trigger scripts

### PR DESCRIPTION
**Pull request by DizzyEggg**
Discussed on Discord https://discord.com/channels/419213663107416084/774393519569502268/1257704663684485261

### CODE REVIEW

File: src/field_control_avatar.c [hunk 1]
```diff
@@ -932,6 +932,16 @@ static s8 GetWarpEventAtPosition(struct MapHeader *mapHeader, u16 x, u16 y, u8 e
...
 }
 
+static bool32 ShouldTriggerScriptRun(const struct CoordEvent *coordEvent)
+{
+    u16 *varPtr = GetVarPointer(coordEvent->trigger);
+    // Treat non Vars as flags
+    if (varPtr == NULL)
+        return (FlagGet(coordEvent->trigger) == coordEvent->index);
+    else
+        return (*varPtr == coordEvent->index);
```
Comment (Bassoonian) [inline src/field_control_avatar.c:942]: Do we not need the original `(u8)` here?

Comment (DizzyEggg) [reply src/field_control_avatar.c:942]: I don't think so. If anything, this only limits var values to 0-0xFF instead of checking 0-0xFFFF

**Comment by tertu-m** [comment]
@Bassoonian any thoughts on this?

**Review by AlexOn1ine — APPROVED** [review]

**Comment by AlexOn1ine** [comment]
> @Bassoonian any thoughts on this?

I've tested it in my hack. Works

**Pull request merged**